File: python/licenselynx/license_object.py
#
# SPDX-FileCopyrightText: Copyright 2025 Siemens AG
# SPDX-License-Identifier: BSD-3-Clause
#
from dataclasses import dataclass
from licenselynx.license_source import LicenseSource


@dataclass
class LicenseObject(object):
    """LicenseObject class represents a license with a canonical name and source."""

    id: str
    src: LicenseSource

    def is_spdx_identifier(self):
        """
        Checks if source of LicenseObject is SPDX.
        :return: True if source of LicenseObject is SPDX
        """
        return self.src == LicenseSource.SPDX

    def is_scancode_licensedb_identifier(self):
        """
        Checks if source of LicenseObject is ScanCode LicenseDB.
        :return: True if source of LicenseObject is ScanCode LicenseDB
        """
        return self.src == LicenseSource.SCANCODE_LICENSEDB

    def is_custom_identifier(self):
        """
        Checks if source of LicenseObject is custom.
        :return: True if source of LicenseObject is custom
        """
        return self.src == LicenseSource.CUSTOM

File: data/licenselynx-licenselynx/go/license_object.go
// SPDX-FileCopyrightText: Copyright 2026 Siemens AG
// SPDX-License-Identifier: BSD-3-Clause

package licenselynx

// LicenseObject represents a canonical license identifier and its source.
type LicenseObject struct {
	ID  string
	Src string
}

// IsSpdxId reports whether the canonical identifier comes from SPDX.
func (l LicenseObject) IsSpdxId() bool {
	return l.Src == string(SourceSPDX)
}

// IsScancodeLicenseId reports whether the canonical identifier comes from ScanCode LicenseDB.
func (l LicenseObject) IsScancodeLicenseId() bool {
	return l.Src == string(SourceScancodeLicenseDB)
}

// IsCustomId reports whether the canonical identifier comes from the custom namespace.
func (l LicenseObject) IsCustomId() bool {
	return l.Src == string(SourceCustom)
}

// IsOrganizationSource reports whether the canonical identifier comes from any organization namespace.
func (l LicenseObject) IsOrganizationSource() bool {
	_, ok := generatedOrgMaps[Organization(l.Src)]
	return ok
}

// IsOrganizationSourceOf reports whether the canonical identifier comes from the provided organization namespace.
func (l LicenseObject) IsOrganizationSourceOf(org Organization) bool {
	return l.Src == string(org)
}
